src/diff/children.js

```diff
@@ -63,16 +63,22 @@ export function diffChildren(parentDom, newParentVNode, oldParentVNode, context,
 		// Check if we find a corresponding element in oldChildren and store the
 		// index where the element was found.
 		p = oldChildren[i];
-		if (p != null && (childVNode.key==null && p.key==null ? (childVNode.type === p.type) : (childVNode.key === p.key))) {
-			index = i;
-		}
-		else {
-			for (j=0; j<oldChildrenLength; j++) {
-				p = oldChildren[j];
-				if (p!=null) {
-					if (childVNode.key==null && p.key==null ? (childVNode.type === p.type) : (childVNode.key === p.key)) {
-						index = j;
-						break;
+
+		if (childVNode!=null) {
+			if (p != null && (childVNode.key==null && p.key==null ? (childVNode.type === p.type) : (childVNode.key === p.key))) {
+				index = i;
+			}
+			else {
+				for (j=0; j<oldChildrenLength; j++) {
+					p = oldChildren[j];
+					if (p!=null) {
+						if (childVNode.key==null && p.key==null
+							? (childVNode.type === p.type && (p._component==null || (p.type.prototype.render==null && p._component.__hooks==null)))
```

Comment:
After a few discussions on slack this line seems wrong when re-redering stateful components where nothing has changed. Need to find some tests for that.

Reply:
It's gone :tada: